src/indicators/drought/chirps_rainfallobservations.py

```diff
@@ -214,63 +239,94 @@ def compute_seasonal_lowertercile_raster(
     )
     (
-        chirps_seasonal_lowertercile_country_filepath,
-        chirps_seaonal_tercile_bounds_country_filepath,
-    ) = get_filepath_seasonal_lowertercile_raster(country_iso3, config)
+        chirps_seasonal_tercile_filepath,
+        chirps_seasonal_tercile_bounds_filepath,
+    ) = get_filepath_seasonal_tercile_raster(country_iso3, config)
 
-    if use_cache and chirps_seasonal_lowertercile_country_filepath.exists():
+    if use_cache and chirps_seasonal_tercile_filepath.exists():
         logger.debug(
-            f"{chirps_seasonal_lowertercile_country_filepath} already exists"
+            f"{chirps_seasonal_tercile_filepath} already exists"
             " and cache is set to True, skipping"
         )
-        return chirps_seasonal_lowertercile_country_filepath
+        return chirps_seasonal_tercile_filepath
 
-    Path(chirps_seasonal_lowertercile_country_filepath.parent).mkdir(
+    Path(chirps_seasonal_tercile_filepath.parent).mkdir(
         parents=True, exist_ok=True
     )
     logger.debug("Computing lower tercile values...")
+    # for some unknown reason the rolling sum takes very long to
+    # compute when using rioxarray so sticking to xarray for now
     ds = xr.load_dataset(chirps_monthly_country_filepath)
+    da = ds.precip
     # compute the rolling sum over three month period. Rolling sum works
     # backwards, i.e. value for month 3 is sum of month 1 till 3. So
     # month==1 is NDJ season
-    ds_season = (
-        ds.rolling(time=seas_len, min_periods=seas_len)
+    da_season = (
+        da.rolling(time=seas_len, min_periods=seas_len)
         .sum()
         .dropna(dim="time", how="all")
     )
     # define the years that are used to define the climatology. We use
     # 1982-2010 since this is also the period used by IRI's seasonal
     # forecasts see
     # https://iri.columbia.edu/our-expertise/climate/forecasts/seasonal-climate-forecasts/methodology/
-    ds_season_climate = ds_season.sel(
-        time=ds_season.time.dt.year.isin(range(1982, 2011))
+    da_season_climate = da_season.sel(
+        time=da_season.time.dt.year.isin(range(1982, 2011))
     )
     # compute the thresholds for the lower tercile, i.e. below average,
     # per season since we computed a rolling sum, each month represents
     # a season
-    ds_season_climate_quantile = ds_season_climate.groupby(
-        ds_season_climate.time.dt.month
-    ).quantile(0.33)
-    # save below tercile boundaries
-    ds_season_climate_quantile.to_netcdf(
-        chirps_seaonal_tercile_bounds_country_filepath
+    da_season_climate_quantile = da_season_climate.groupby(
+        da_season_climate.time.dt.month
+    ).quantile([1 / 3, 2 / 3], skipna=True)
+
+    # save tercile boundaries
+    da_season_climate_quantile.to_netcdf(
+        chirps_seasonal_tercile_bounds_filepath
+    )
+
+    da_bn = xr.where(
+        da_season
+        <= da_season_climate_quantile.sel(
+            quantile=1 / 3, month=da_season.time.dt.month
```

Comment:
Is there a way to rename the quantile variables? It's just that selecting on a float value seems like it's asking for trouble

Reply:
Great idea, renamed them to `lower_bound` and `upper_bound`